storage/storage_image.go

```diff
@@ -427,23 +777,58 @@ func (s *storageImageDestination) SupportedManifestMIMETypes() []string {
 	return manifestMIMETypes
 }
 
-// PutManifest writes manifest to the destination.
-// FIXME? This should also receive a MIME type if known, to differentiate between schema versions.
-// If the destination is in principle available, refuses this manifest type (e.g. it does not recognize the schema),
-// but may accept a different manifest type, the returned error must be an ManifestTypeRejectedError.
+// GetManifest reads the manifest that we intend to store.  If we haven't been given one (yet?),
+// generate one.
+func (s *storageImageUnnamedDestination) GetManifest() ([]byte, string, error) {
+	if len(s.named.manifest) == 0 {
```

Comment:
The pathological case, I guess.  If `Commit()` is called without a manifest having been written, we would fail, but if that's what's expected for that case, it'll probably simplify things.